## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('/Users/yuliias/Documents/Data analysis/Course/Home Works/data_statistics/data.csv')

In [2]:
data = df['data'].values
type(data)

numpy.ndarray

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [3]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [29]:
!pip3 install statsmodels


[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: pip3 install --upgrade pip


In [4]:
from statsmodels.stats.weightstats import ztest

In [5]:
null_mean = 20
alpha = 0.05

In [6]:
np.mean(data)

np.float64(19.37755319670441)

In [7]:
ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')

In [8]:
ztest_Score, p_value

(np.float64(-0.9953477389335053), np.float64(0.8402164165464012))

In [9]:
if p_value < 0.05:
    print("Відхиляємо H0: середній чек статистично значуще відрізняється від $20.")
else:
    print("Не відхиляємо H0: середній чек статистично НЕ відрізняється від $20.")

Не відхиляємо H0: середній чек статистично НЕ відрізняється від $20.


2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [10]:
df = pd.read_csv('/Users/yuliias/Documents/Data analysis/Course/Home Works/data_statistics/data_2.csv')

In [11]:
data_2 = df['data'].values

In [12]:
data_2.shape

(120,)

In [13]:
np.mean(data), np.mean(data_2)

(np.float64(19.37755319670441), np.float64(19.37322276512519))

In [14]:
ztest_Score_2, p_value_2 = ztest(data, data_2, value=0)

In [15]:
ztest_Score_2, p_value_2

(np.float64(0.005246673456808504), np.float64(0.9958137794593356))

In [16]:
if p_value < 0.05:
    print("Є статистично значуща різниця у середніх значення чеку в липні та серпні 2022 року.")
else:
    print("Немає статистично значущої різниці у середніх значення чеку в липні та серпні 2022 року.")

Немає статистично значущої різниці у середніх значення чеку в липні та серпні 2022 року.


In [19]:
print(f"Середнє значення за липень 2022 = {np.mean(data)}")
print(f"Середнє значення за серпень 2022 = {np.mean(data_2)}")
print(f"Різниця середніх складає = {np.mean(data_2) - np.mean(data)}")

Середнє значення за липень 2022 = 19.37755319670441
Середнє значення за серпень 2022 = 19.37322276512519
Різниця середніх складає = -0.004330431579223415


Бачимо, що різниця між середніми значеннями дуже маленька, а одже можемо вважати їх рівними.
За результатами двовибіркового тесту за серпень 2022 бачимо, що p_value_2 = 0.995 більше за 0.05, тому відкинули гіпотезу. Ми не маємо статистично значущої різниці.